test/unit/CustomFunction.spec.ts

```diff
@@ -91,88 +56,40 @@ describe('CustomFunction class', () => {
+      // intentionally casting to Middleware<SlackCustomFunctionMiddlewareArgs>[] to trigger failure
+      const badMiddleware = [
+        async () => {},
+        'not-a-function',
+      ] as unknown as Middleware<SlackCustomFunctionMiddlewareArgs>[];
 
       const validationFn = () => validate('callback_id', badMiddleware);
       const expectedMsg = 'All CustomFunction middleware must be functions';
       assert.throws(validationFn, CustomFunctionInitializationError, expectedMsg);
     });
   });
 
-  describe('isFunctionEvent', () => {
-    it('should return true if recognized function_executed payload type', async () => {
-      const fakeExecuteArgs = createFakeFunctionExecutedEvent();
-
-      const { isFunctionEvent } = await importCustomFunction();
-      const eventIsFunctionExcuted = isFunctionEvent(fakeExecuteArgs);
-
-      assert.isTrue(eventIsFunctionExcuted);
-    });
-
-    it('should return false if not a function_executed payload type', async () => {
-      const fakeExecutedEvent = createFakeFunctionExecutedEvent();
-      // @ts-expect-error expected invalid payload type
-      fakeExecutedEvent.payload.type = 'invalid_type';
-
-      const { isFunctionEvent } = await importCustomFunction();
-      const eventIsFunctionExecuted = isFunctionEvent(fakeExecutedEvent);
-
-      assert.isFalse(eventIsFunctionExecuted);
-    });
-  });
-
-  describe('enrichFunctionArgs', () => {
-    it('should remove next() from all original event args', async () => {
-      const fakeExecutedEvent = createFakeFunctionExecutedEvent();
-
-      const { enrichFunctionArgs } = await importCustomFunction();
-      const executeFunctionArgs = enrichFunctionArgs(fakeExecutedEvent, {});
-
-      assert.notExists(executeFunctionArgs.next);
-    });
-
-    it('should augment function_executed args with inputs, complete, and fail', async () => {
-      const fakeArgs = createFakeFunctionExecutedEvent();
-
-      const { enrichFunctionArgs } = await importCustomFunction();
-      const functionArgs = enrichFunctionArgs(fakeArgs, {});
-
-      assert.exists(functionArgs.inputs);
-      assert.exists(functionArgs.complete);
-      assert.exists(functionArgs.fail);
-    });
-  });
-
   describe('custom function utility functions', () => {
     describe('`complete` factory function', () => {
       it('complete should call functions.completeSuccess', async () => {
         const client = new WebClient('sometoken');
         const completeMock = sinon.stub(client.functions, 'completeSuccess').resolves();
-        const complete = CustomFunction.createFunctionComplete(
-          { isEnterpriseInstall: false, functionExecutionId: 'Fx1234' },
-          client,
-        );
+        const complete = createFunctionComplete({ isEnterpriseInstall: false, functionExecutionId: 'Fx1234' }, client);
         await complete();
         assert(completeMock.called, 'client.functions.completeSuccess not called!');
```

Comment:
```suggestion
        assert(completeMock.called, 'client.functions.completeSuccess not called!');
        assert(
          completeMock.calledWith({ function_execution_id: 'Fx1234', outputs: {} }),
          'client.functions.completeSuccess called with unexpected arguments!',
        );
```

IMO checking that the right arguments are used would be a nice confidence boost!